lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -1852,10 +1852,26 @@ sub search_for {
     return $self->$result_set->search($condition, $attrs);
 }
 
+# if the job is not blocked by chains or parallels
+sub is_edge_in_cluster {
+    my ($self, $cluster_info) = @_;
+
+    for my $key (qw(parallel_children parallel_parents chained_parents)) {
+        return 0 if scalar(@{$cluster_info->{$self->id}->{$key}}) > 0;
+    }
+    return 1;
+}
+
 sub blocked_by_parent_job {
     my ($self) = @_;
 
     my $cluster_jobs = $self->cluster_jobs;
+
+    # chained parents are part of the cluster, but don't
+    # normally become blocked_by, so make this extra step
+    return undef if $self->is_edge_in_cluster($cluster_jobs);
+
+    # now the complicated part
```

Comment:
I thought it was 'simplified' scheduler :)

Reply:
It's complicated if you touch it before 7am - still simple :)